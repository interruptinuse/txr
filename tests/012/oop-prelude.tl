(load "../common")

(define-struct-prelude init-fini-log (fox bear)
  (:init (me) (put-line `@me created`))
  (:fini (me) (put-line `@me finalized`)))

(defstruct fox ())

(defstruct bear ())

(with-objects ((f (new fox))
               (b (new bear)))
  (put-line "inside with-object"))
