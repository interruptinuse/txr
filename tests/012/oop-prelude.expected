#S(fox) created
#S(bear) created
inside with-object
#S(bear) finalized
#S(fox) finalized
